convert2rhel/subscription.py

```diff
@@ -179,7 +180,9 @@ def register_system():
         registration_cmd = RegistrationCommand.from_tool_opts(tool_opts)
 
         try:
+            os_release_file.restore(rollback=False)  # RHELC-16
             registration_cmd()
+            os_release_file.remove()  # RHELC-16
```

Comment:
Add a comment why we need to do this. If I were to look at this code after a while I wouldn't know why we're doing it without going through an issue